hedera-node/src/main/java/com/hedera/services/txns/ethereum/EthTxData.java

```diff
@@ -0,0 +1,237 @@
+package com.hedera.services.txns.ethereum;
+
+import com.esaulpaugh.headlong.rlp.RLPDecoder;
+import com.esaulpaugh.headlong.rlp.RLPEncoder;
+import com.esaulpaugh.headlong.rlp.RLPItem;
+import com.esaulpaugh.headlong.util.Integers;
+import com.google.common.base.MoreObjects;
+import com.hedera.services.state.submerkle.EvmFnResult;
+import org.apache.commons.codec.binary.Hex;
+import org.bouncycastle.jcajce.provider.digest.Keccak;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+
+public record EthTxData(
+		byte[] rawTx,
+		EthTransactionType type,
+		byte[] chainId,
+		long nonce,
+		byte[] gasPrice,
+		byte[] maxPriorityGas,
+		byte[] maxGas,
+		long gasLimit,
+		byte[] to,
+		BigInteger value,
+		byte[] callData,
+		byte[] accessList,
+		int recId,
+		byte[] v,
+		byte[] r,
+		byte[] s) implements EvmFnResult.EvmFnCallContext {
+
+	static final BigInteger WEIBARS_TO_TINYBARS = BigInteger.valueOf(10_000_000_000L);
+
+	// TODO constants should be in besu-native
+	static final int SECP256K1_FLAGS_TYPE_COMPRESSION = 1 << 1;
+	static final int SECP256K1_FLAGS_BIT_COMPRESSION = 1 << 8;
+	static final int SECP256K1_EC_COMPRESSED = (SECP256K1_FLAGS_TYPE_COMPRESSION | SECP256K1_FLAGS_BIT_COMPRESSION);
+
+	public static EthTxData populateEthTxData(byte[] data) {
+		var decoder = RLPDecoder.RLP_STRICT.sequenceIterator(data);
+		var rlpItem = decoder.next();
+		EthTransactionType type;
+		long nonce;
+		byte[] chainId = null;
+		byte[] gasPrice = null;
+		byte[] maxPriorityGas = null;
+		byte[] maxGas = null;
+		long gasLimit;
+		byte[] to;
+		BigInteger value;
+		byte[] callData;
+		byte[] accessList = null;
+		byte recId;
+		byte[] v = null;
+		byte[] r;
+		byte[] s;
+		if (rlpItem.isList()) {
+			// frontier TX
+			type = EthTransactionType.LEGACY_ETHEREUM;
+			List<RLPItem> rlpList = rlpItem.asRLPList().elements();
+			if (rlpList.size() != 9) {
+				return null;
+			}
+			nonce = rlpList.get(0).asLong();
+			gasPrice = rlpList.get(1).asBytes();
```

Comment:
Per [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559), should maxPriorityGas and maxGas be set to gasPrice for legacy transactions when they normalize the transaction?